commitlint.config.js

```diff
@@ -0,0 +1,135 @@
+module.exports = {
```

Comment:
Can the file be called `.commitlint.config.js`?  I'm concerned that some of our community members will object to having a "visible" config file at the top level.  Plus, we have logic in the spec file to strip out unnecessary dot files which will not pick up this file, or we will have to add a special case for it